Source/WebCore/dom/ContentVisibilityDocumentState.cpp

```diff
@@ -123,10 +125,13 @@ DidUpdateAnyContentRelevancy ContentVisibilityDocumentState::updateRelevancyOfCo
             if (relevancyToCheck.contains(ContentRelevancy::IsInTopLayer))
                 setRelevancyValue(ContentRelevancy::IsInTopLayer, hasTopLayerinSubtree(*target));
 
-            if (oldRelevancy == newRelevancy)
+            if (oldRelevancy == newRelevancy && oldRelevancy)
```

Comment:
Why is this correct? If both old & new are both empty then there is no event to be emitted right?

Reply:
The spec is not too clear about it, but both gecko and chromium dispatch this event when first determining rendering state, independent whether that turns out skipped or not. To make sure we are catching the first rendering state case I made the content relevancy std::optional.

Reply:
Can we add a test for it?

Reply:
Oh I guess that's imported/w3c/web-platform-tests/css/css-contain/content-visibility/content-visibility-auto-state-changed-first-observation.html

Reply:
Correct :)